apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -310,6 +386,45 @@ Gallery.prototype = {
     }.bind(this));
   },
 
+  switchToGalleryFrame: function() {
+    var galleryFrame = this.client.findElement('iframe[src*="' +
+                                          Gallery.ORIGIN + '"]');
```

Comment:
Does this.client.findElement() wait for the element to be created, or does it fail if it does not exist when called?

If the gallery is being launched by an activity, it seems like there might be intermittent test failures because of variable timing if we're not really careful here.